core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/signup.jelly

```diff
@@ -73,7 +73,7 @@ THE SOFTWARE.
                   ${data.errors.containsKey('username') ? data.errors.get('username') : ''}
                 </div>
                 <input class="jenkins-input ${data.errors.containsKey('username') ? 'jenkins-input--error' : ''}"
-                       autofocus="${(data.errors.containsKey('username') || !data) ? true : null}"
+                       autofocus="${(data == null || data.errors.containsKey('username')) ? true : null}"
```

Comment:
the `null` check needs to occur first as it will be null if this is the first visit (rather than a return because some information was incorrect).